scripts/download.mts

```diff
@@ -0,0 +1,161 @@
+    if(lastProgressStr !== progressStr) {
+      readline.cursorTo(process.stdout, 0);
+      process.stdout.write(progressStr);
+      readline.clearLine(process.stdout, 1);
+      lastProgressStr = progressStr;
+    }
+  });
+  // add new line after download percentage
+  process.stdout.write('\n');
+
+  fs.appendFileSync(destFile, Buffer.from(downloadResponse.rawBody));
+  console.log(`Downloaded to ${destFile}`);
+
+  console.log(`Verifying ${destFile}...`);
+
+  if (!(await checkFileSha(destFile, fileSha))) {
+    console.warn(`Checksum for downloaded ${destFile} is not match, downloading again...`);
+    fs.rmSync(destFile);
+    downloadAttempt++;
+    downloadAndCheckSha(downloadUrl, destFile, fileSha);
+  } else {
+    console.log(`Checksum for ${destFile} is matched.`);
+  }
+
+}
+
+async function downloadCrc(): Promise<void> {
+
+  console.info(`Found CRC: ${bundledCrc.version.crcVersion}`);
+
+
+  let releaseUrl: string;
+
+  if(platform === 'win32'){
+    releaseUrl = bundledCrc.links.windows;
+  } else if(platform === 'darwin') {
+    releaseUrl = bundledCrc.links.darwin;
+  } else {
+    releaseUrl = bundledCrc.links.linux;
+  }
+
+  console.info('Download SHA sums...');
+  const shaSumUrl = releaseUrl.substring(0, releaseUrl.lastIndexOf('/')) + '/sha256sum.txt';
+  const shaSumContentResponse = await got.get(shaSumUrl);
+
+  const installerFileName = platformToInstallFileName[platform];
+  if(!installerFileName) {
+    console.error('Can\'t find installer file name. Exiting...');
+    process.exit(1);
+  }
+
+
+  const shasSumArr = shaSumContentResponse.body.split('\n');
+
+  let installerSha = '';
+  for(const shaLine of shasSumArr){
+    if(shaLine.trim().endsWith(installerFileName)) {
+      installerSha = shaLine.split(' ')[0];
+      break;
+    }
+  }
+
+  if(!installerSha) {
+    console.error(`Can't find SHA256 sum for ${installerFileName} in:\n${shaSumContentResponse.body}`);
+    process.exit(1);
+  }
+
+  const destDir = path.resolve(__dirname, '..', 'assets');
+  if (!fs.existsSync(destDir)) {
+    fs.mkdirSync(destDir);
+  }
+  const destFile = path.resolve(destDir, installerFileName);
+
+  if (!fs.existsSync(destFile)) {
+    console.log(`Downloading Podman package from ${releaseUrl}`);
+    downloadAndCheckSha(releaseUrl, destFile, installerSha);
+    return;
+  } else {
+    console.log(`Podman package ${releaseUrl} already downloaded.`);
+  }
+
+  console.log(`Verifying ${installerFileName}...`);
+
+  if (!(await checkFileSha(destFile, installerSha))) {
+    console.warn(`Checksum for downloaded ${destFile} is not match, downloading again...`);
+    fs.rmSync(destFile);
+    downloadAttempt++;
+    downloadAndCheckSha(releaseUrl, destFile, installerSha);
+  } else {
+    console.log(`Checksum for ${installerFileName} is matched.`);
+  }
+
+}
+
+
+if(platform === 'win32' /*|| platform === 'darwin'*/){
+  downloadCrc();
```

Comment:
This should be done on initialize of the extension, so after the GUI is available.

Reply:
Actually, this story is not defined... but it would make sense to allow downloading of 'the latest' version from the extension prior (or part of) the actual install.